tests/tracer/test_tracer.py

```diff
@@ -1514,3 +1514,49 @@ def test_spans_sampled_all(tracer, test_spans):
 
     spans = test_spans.pop()
     assert len(spans) == 3
+
+
+def test_closing_other_context_spans_single_span(tracer, test_spans):
```

Comment:
curiosity question, what happens with processes?

since when we fork we'll have the same open span already in memory? or do the contextvars get reset?

Reply:
It's worth having test cases for processes as well.

Yes, this is handled as the context is cloned (clearing the span reference) if a fork is detected when starting a span. This might actually be a case that is missed in #1936 though :eyes:

```python
span = tracer.trace(...)
pid = os.fork()

if not pid:
  # finish in parent
  span.finish()
else:
  # child still open
  ...
```

Reply:
Adding this test case caught a bug! #2497